module-2/backend/week_one.md

```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
   
   ```ruby
   get '/horses' do
+  @count = 1
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+
+  ```ruby
+  get '/horses/:name' do
+    erb :index, :locals => {:name => params[:name]}
+  end
+  ```
+
 9. What's the purpose of ERB?
+
+        It allows you to embedd Ruby into a text or HTML document.
+
 10. Why do I need a development AND test database?
+
+        There are two databases so tests that are run do not corrupt the data in your development database.
+
 11. What's responsive design?
+
+        It's an approach to web design aimed at allowing desktop webpages to be viewed in response to the size of the screen or web browser someone is viewing with. 
+
 12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
+
+        It's an acronym for CREATE, READ, UPDATE, and DELETE. These are the major functions that are implemented in relational database applications.
+
+13. What does HTTP stand for?
+
+        Hypertext Transfer Protocal.
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+        <% %> Runs the Ruby inside of it but doesn't display it
+        <%= %> Displays the Ruby that is being run inside of it
+
 15. What's an ORM?
+
+        Object Relational Mapping. It connects the rich objects of an application to tables in a relational database management system. The properties and relationships of the objects in an application can be easily stored and retrieved from a database without writing SQL statements directly
+
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+
+        Active Record
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
+
+get '/restaurants'
+
+        Displays a list of all restaurants.
+
+get '/restaurants/new'
+
+        Returns an HTML form for creating a new restaurant.
+
+post '/restaurants'
+
+        Creates a new restaurant.
+
+get '/restaurants/:id/edit'
+
+        Returns an HTML form for editing a restaurant.
+
+put '/restaurants/:id'
+
+        Updates a specific restaurant.
+
+get '/restaurants/:id'
+
+        Displays a specific restaurant.
+
+delete '/restaurants/:id'
+
+        Deletes a specific restaurant.
+
+18. What's a migration?
+
+        A migration alters your database schema over time. Each migration modifies your schema to add or remove tables, columns, or entries.
+
 19. When you create a migration, does it automatically modify your database?
+
+        I believe it only modifies your database if you format your migration table for new changes.

```

Comment:
True to some degree. The more important point for us at this time is that migrations don't change your database unless you run `rake db:migrate`.